packages/framework/src/Model/Product/ProductDataFactory.php

```diff
@@ -124,12 +124,23 @@ public function createFromProduct(Product $product)
         $productData = $this->createDefault();
 
         $translations = $product->getTranslations();
+        $domains = $product->getDomains();
         $names = [];
         $variantAliases = [];
         foreach ($translations as $translation) {
             $names[$translation->getLocale()] = $translation->getName();
             $variantAliases[$translation->getLocale()] = $translation->getVariantAlias();
         }
+
+        foreach ($domains as $domain) {
```

Comment:
We usually call instances of DomainConfig a $domain, these objects should be called $productDomains imo to avoid confusion.